### PR TITLE
FUTURE(ext/net): remove `Deno.ListenTlsOptions.(keyFile|certFile)`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #23089

### CODE REVIEW

File: ext/net/02_tls.js [hunk 1]
```diff
@@ -151,13 +151,23 @@ function listenTls({
       new Error().stack,
       "Pass the key file contents to the `Deno.ListenTlsOptions.key` option instead.",
     );
+    if (internals.future) {
+      throw new TypeError(
+        "`Deno.ListenTlsOptions.keyFile` has been removed in favor of `Deno.ListenTlsOptions.key`",
+      );
+    }
```
Comment (bartlomieju) [inline ext/net/02_tls.js:158]: Maybe instead of throwing an error we just make it `undefined` so it doesn't work (as would happen in Deno 2)

Comment (iuioiua) [reply ext/net/02_tls.js:158]: I remember seeing some unsupported option throwing when used somewhere in the Deno codebase. I can't find where that was, so, yep, I will revert.

**Comment by iuioiua** [comment]
This passes on my local machine. Not sure why it's failing in CI.

**Comment by iuioiua** [comment]
@mmastrac, I'm setting the deprecated cert and key properties to `undefined` if `DENO_FUTURE=1` is being used. Now, we're panicking because we're hitting https://github.com/denoland/deno/blob/9304126be5633d4e7d384a8df87f5833a7a145e2/ext/net/ops_tls.rs#L452. Seeing as you made some changes around here in #23296, how do you suggest we handle this? Perhaps, instead of returning `op_tls_key_null()`, we could throw some error saying that none of the cert or key properties are defined but need to be. This will also affect #23270.

**Review by bartlomieju — APPROVED** [review]
LGTM, please rebase

**Pull request merged**